functions/connections.cpp

```diff
@@ -7,17 +7,23 @@
 #include <any>
 #include "../data/types.h"
 
-std::shared_ptr<std::map<std::string, std::any>> tinygraph_vertex_link(const std::shared_ptr<Vertex>& from, const std::shared_ptr<Vertex>& to, bool undirected) {
-    auto to_edge = std::shared_ptr<Edge>(new Edge(to));
-    to_edge->properties = std::shared_ptr<std::map<std::string, std::any>>(new std::map<std::string, std::any>);
-    from->connections.push_back(to_edge);
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "performance-unnecessary-value-param"
```

Comment:
Done, sort of (?)

Reply:
I think that's probably fine, though if you don't want the move it should be alright to make `from` a const reference because it isn't necessarily copied in every code path.

Reply:
Done